mllib/src/main/scala/org/apache/spark/mllib/tree/DecisionTree.scala

```diff
@@ -1064,9 +1045,12 @@ object DecisionTree extends Serializable with Logging {
               //   Bins correspond to feature values, so we do not need to compute splits or bins
               //   beforehand.  Splits are constructed as needed during training.
               splits(featureIndex) = new Array[Split](0)
-              bins(featureIndex) = new Array[Bin](0)
             }
-          }
+            // For ordered features, bins correspond to feature values.
+            // For unordered categorical features, there is no need to construct the bins.
+            // since there is a one-to-one correspondence between the splits and the bins.
+            bins(featureIndex) = new Array[Bin](0)
+            }
```

Comment:
The closing brace should be 2 spaces to the left: [https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide]